### ISSUE TITLE
More the Max Quantity Added to Cart

### ISSUE THREAD

**Issue by chris-huffman** [magento/magento2#28517]
### Preconditions (*)

1. Magento 2.3.4
2. The following was reproduced on a fresh install with sample data.

### Steps to reproduce (*)

1. Configure a simple product with a max quantity. 
2. Login to the frontend.
3. Add the max quantity of the configured item to your cart.
4. Logout.
5. As a guest, add max quantity of the item to your cart. (Or any quantity.)
6. Log back in as your user.
7. During the quote merge process, the item quantities are combined, and the result is more quantity in the cart than should be allowed.

### Expected result (*)

1. When the quotes are merged, I would expect the max quantity to be in the cart.

### Actual result (*)

1. When quotes are merged, the quantity is combined, resulting in a quantity greater than the max quantity configured on the product.
2. [Screenshot](https://files.classyllama.com/6cf7c12c/Joust_Duffle_Bag_and_Joust_Duffle_Bag___Products___Inventory___Catalog___Magento_Admin_24895616.png)

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by swathishs** [comment]
@magento give me 2.4-develop instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.4-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @swathishs, here is your [Magento instance](https://i-28517-2-4-develop.instances.magento-community.engineering).
Admin access: [https://i-28517-2-4-develop.instances.magento-community.engineering/admin_d20e](https://i-28517-2-4-develop.instances.magento-community.engineering/admin_d20e)
Login: `71411cb4` Password: `51f0f12d92da`
Instance will be terminated in up to 3 hours.

**Comment by swathishs** [comment]
@chris-huffman: Tested the issue in Magento 2.4 develop instance. We can reproduce it but validation there in mini cart page so this bug is not valid in Magento 2.4 develop instance 

Please find the attachment below:

1. Admin:
![image](https://user-images.githubusercontent.com/58288685/83788772-56923280-a6b3-11ea-8082-198eed33d31c.png)
2. Mini cart:
![image](https://user-images.githubusercontent.com/58288685/83788803-6578e500-a6b3-11ea-882a-6e20e504a423.png)
3. Cart page:
![image](https://user-images.githubusercontent.com/58288685/83788822-6f024d00-a6b3-11ea-8135-137d251d4d26.png)

**Comment by swathishs** [comment]
@magento give me 2.3.5-p1 instance

**Comment by magento-engcom-team** [comment]
Hi @swathishs. Thank you for your request. I'm working on Magento 2.3.5-p1 instance for you

**Comment by magento-engcom-team** [comment]
Hi @swathishs, here is your [Magento instance](https://i-28517-2-3-5-p1.instances.magento-community.engineering).
Admin access: [https://i-28517-2-3-5-p1.instances.magento-community.engineering/admin_3c47](https://i-28517-2-3-5-p1.instances.magento-community.engineering/admin_3c47)
Login: `b6322bf6` Password: `a08859cf49e0`
Instance will be terminated in up to 3 hours.

**Comment by swathishs** [comment]
@chris-huffman: Tested the issue on 2.3.5-p1 instance the issue is reproducible but there is a validation on cart page:

1. Admin:
![image](https://user-images.githubusercontent.com/58288685/83790910-8b53b900-a6b6-11ea-8080-0940d58223ef.png)
2. Mini cart:
![image](https://user-images.githubusercontent.com/58288685/83790944-97d81180-a6b6-11ea-8569-e7a0d775f016.png)
3. cart page:
![image](https://user-images.githubusercontent.com/58288685/83790969-a292a680-a6b6-11ea-8381-c180414b3e4b.png)

**Comment by chris-huffman** [comment]
@swathishs thanks for looking into this! 

I also saw the validation on the cart page (in 2.3.4). I guess it's easier to add a message than it is to fix this occurrence from happening? Looks like 2.4 fixes the message not displaying in the minicart. I think showing this message in 2.3.4 and 2.3.5 would be enough for me to consider this resolved. Right now I see [this](https://files.classyllama.com/6cf7c12c/Shopping_Cart_24896C27.png)

**Comment by swathishs** [comment]
@chris-huffman: Shall I close this issue. Or should I open the issue occurs only in 2.3.5 p1

**Comment by chris-huffman** [comment]
@swathishs  I would like to see the minicart message fixed in 2.3.5 p1. I think that would help the user experience here.

**Comment by swathishs** [comment]
Can be reproduced only on Magento 2.3.5p1 but not on Magento 2.4 develop instance

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @swathishs 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-34959` were created

**Issue Available**: @swathishs, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by sdzhepa** [comment]
Hello @chris-huffman

Thank you for your report and collaboration!

Unfortunately, we are closing this issue as low priority and as fixed in the current `2.4-develop`

According to [Code Contributions](https://devdocs.magento.com/contributor-guide/contributing.html#backlog) requirements and section [Accepted pull requests and ported code](https://devdocs.magento.com/contributor-guide/contributing.html#rules)
> v2.4 Contribution Notice: Magento is focusing development efforts on v2.4, so we are no longer accepting pull requests to the v2.3, v2.2, v2.1, and v2.0 release lines.

Only `2.4-develop` brunch is opened for community contribution and collaboration.
All code changes to `2.3.x` release line can be done by internal teams only.
It supposed to be very critical issues only and only after internal approval per each based on the prioritized backlog.
The goal is to minimize all quality risks and regression testing scope.

_Note: If you are an Enterprise/Commerce user with support assigned and this issue critical for your store, I would recommend to use the Support portal associated with your account._
